cpp/src/text/tokenize.cu

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/utilities/error.hpp>
+#include <nvtext/tokenize.hpp>
+#include <nvtext/detail/tokenize.hpp>
+#include <text/utilities/tokenize_ops.cuh>
+
+#include <thrust/transform.h>
+
+namespace nvtext
+{
+namespace detail
+{
+namespace
+{
+
+// common pattern for token_count functions
+template<typename TokenCounter>
+std::unique_ptr<cudf::column> token_count_fn( cudf::size_type strings_count, TokenCounter tokenizer,
+                                              rmm::mr::device_memory_resource* mr,
+                                              cudaStream_t stream )
+{
+    // create output column
+    auto token_counts = cudf::make_numeric_column( cudf::data_type{cudf::INT32}, strings_count,
+                                                   cudf::UNALLOCATED, stream, mr);
+    auto d_token_counts = token_counts->mutable_view().data<int32_t>();
+    // add the counts to the column
+    thrust::transform( rmm::exec_policy(stream)->on(stream),
+        thrust::make_counting_iterator<cudf::size_type>(0),
+        thrust::make_counting_iterator<cudf::size_type>(strings_count),
+        d_token_counts, tokenizer );
+    return token_counts;
+}
+
+// common pattern for tokenize functions
+template<typename Tokenizer>
+std::unique_ptr<cudf::column> tokenize_fn( cudf::size_type strings_count, Tokenizer tokenizer,
+                                           rmm::mr::device_memory_resource* mr,
+                                           cudaStream_t stream )
+{
+    auto execpol = rmm::exec_policy(stream);
+    // get the number of tokens in each string
+    auto token_counts = token_count_fn( strings_count, tokenizer, mr, stream );
```

Comment:
Can you put a `const` here as well?
```suggestion
    auto const token_counts = token_count_fn( strings_count, tokenizer, mr, stream );
```

Reply:
This variable is of type `std::unique_ptr<cudf::column>` 
It is not being transferred and is automatically deleted so I suppose it can be `const`.

Reply:
👍